hip_data_tools/apache/cassandra.py

```diff
@@ -221,21 +290,35 @@ def upsert_dataframe(self, dataframe: DataFrame, table: str) -> None:
         """
         prepared_statement = self._session.prepare(
             self._cql_upsert_from_dataframe(dataframe, table))
-        batch = _prepare_batch(prepared_statement,
-                               _extract_rows_from_dataframe(dataframe))
-        return self._session.execute(batch)
 
-    def upsert_dict(self, data: list, table: str) -> ResultSet:
+        batches = _prepare_batches(prepared_statement, _extract_rows_from_dataframe(dataframe))
+        return self._execute_batches(batches)
+
+    def _execute_batches(self, batches):
+        results = []
```

Comment:
lots of logic for a private method

Reply:
ummm, not a lot of logic in here. if you remove the tqdm which is a form of logging (it creates a progress bad in logs) and the actual logs, this is what you are left with 
```
    def _execute_batches(self, batches):
        results = []
        for batch in batches:
            results.append(self._execute_batch(batch))
        return results
```
 this method does just one thing, i.e. iterate over a collection.
which is equivalent to a list comprehension, I just choose todo it in a for loop to help with logging.